service/src/main/java/uk/gov/hmcts/reform/fpl/selectors/ChildrenSmartSelector.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.reform.fpl.selectors;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.event.ManageOrdersEventData;
+import uk.gov.hmcts.reform.fpl.model.order.OrderTempQuestions;
+import uk.gov.hmcts.reform.fpl.service.ChildrenService;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Component
+@RequiredArgsConstructor
+public class ChildrenSmartSelector {
+
+    private final ChildrenService childrenService;
+
+    public List<Element<Child>> getSelectedChildren(CaseData caseData) {
+        boolean onlyOneChildCanBeSelected = canOnlyOneChildBeSelected(caseData);
+        if (onlyOneChildCanBeSelected) {
+            return getSelectedChildFromSingleSelectionComponent(caseData);
+        } else {
+            return childrenService.getSelectedChildren(caseData);
+        }
+    }
+
+    public static boolean canOnlyOneChildBeSelected(CaseData caseData) {
```

Comment:
That's because they're shared between the SmartSelector and the SmartUpdater.
Happy to put them in a common parent class or a helper class. What do you think?